### PR TITLE
DEV-243: WL Fix auto-installer bugs

### PR DESCRIPTION
**Pull request by luiscuenca**
https://highfidelity.atlassian.net/browse/DEV-243

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20620/HighFidelity-Beta-PR15997-8fde82d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20620/HighFidelity-Beta-Interface-PR15997-8fde82d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20620/HighFidelity-Beta-PR15997-8fde82d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20620/HighFidelity-Beta-Interface-PR15997-8fde82d.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20620/HighFidelity-Beta-PR15997-8fde82d.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20620/HighFidelity-Quest-Beta-PR15997-8fde82d.apk)

File: launchers/win32/LauncherApp.cpp [hunk 1]
```diff
@@ -53,10 +52,10 @@ BOOL CLauncherApp::InitInstance() {
                 restarting = true;
             } else if (curArg.Compare(_T("--noUpdate")) == 0) {
                 noUpdate = true;
-            } else if (curArg.Compare(_T("--continueUpdating")) == 0) {
-                continueUpdating = true;
-            } else if (curArg.Compare(_T("--skipSplash")) == 0) {
-                skipSplash = true;
+            } else if (curArg.Compare(_T("--startScreen")) == 0) {
+                if (i + 1 < iNumOfArgs) {
+                    startScreen = (CLauncherDlg::DrawStep)_wtoi(pArgs[i + 1]);
```
Comment (huffman) [inline launchers/win32/LauncherApp.cpp]: What passes this in?

Comment (huffman) [reply launchers/win32/LauncherApp.cpp]: More specifically, is it the old launcher, and if so, will this be an issue if screen #s change?

Comment (luiscuenca) [reply launchers/win32/LauncherApp.cpp]: This is a way to tell the updated launcher what action should continue performing once is executed. i.e. When we have to install a new launcher and a new interface the updated launcher should execute directly showing de update screen with the progress at 20%. Or when interface is not installed the updated launcher should directly show the login screen, so we can show some continuity. An issue could occur if we change the order of the enums but I think this is unlikely. I forsee maybe changing the names or adding new ones which will not affect the implementation.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20670/HighFidelity-Beta-PR15997-677e021.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20670/HighFidelity-Beta-Interface-PR15997-677e021.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20670/HighFidelity-Beta-PR15997-677e021.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20670/HighFidelity-Beta-Interface-PR15997-677e021.dmg

File: launchers/win32/LauncherApp.cpp (hunk 1 above)
Comment (huffman) [reply launchers/win32/LauncherApp.cpp]: (We talked via chat - thanks for the explanation Luis!)

**Comment by huffman** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20673/HighFidelity-Beta-PR15997-677e021.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20673/HighFidelity-Beta-Interface-PR15997-677e021.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20673/HighFidelity-Beta-PR15997-677e021.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20673/HighFidelity-Quest-Beta-PR15997-677e021.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20684/HighFidelity-Beta-PR15997-677e021.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20684/HighFidelity-Beta-Interface-PR15997-677e021.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20684/HighFidelity-Beta-PR15997-677e021.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20684/HighFidelity-Quest-Beta-PR15997-677e021.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20688/HighFidelity-Beta-PR15997-9fbacd4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20688/HighFidelity-Beta-Interface-PR15997-9fbacd4.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20688/HighFidelity-Beta-PR15997-9fbacd4.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20688/HighFidelity-Quest-Beta-PR15997-9fbacd4.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20688/HighFidelity-Beta-PR15997-9fbacd4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20688/HighFidelity-Beta-Interface-PR15997-9fbacd4.exe

**Pull request merged**